event.go

```diff
@@ -311,24 +313,22 @@ func NewEventFromUntrustedJSON(eventJSON []byte, roomVersion RoomVersion) (resul
 				return
 			}
 		}
-
-		eventJSON = redactedJSON
 	}
 
-	result.eventJSON = eventJSON
-
```

Comment:
Why did you remove these lines?

Reply:
It's now done at the correct time in `populateFieldsFromJSON`, e.g. after removing `event_id` in the new event format.